CSK_Module_MultiIOLinkSMI/scripts/CSK_MultiIOLinkSMI_Processing.lua

```diff
@@ -504,21 +596,29 @@ local function updateIODDReadMessages()
   updateFunctions()
   local queueFunctions = {}
   for messageName, messageInfo in pairs(ioddReadMessages) do
-    if helperFuncs.getTableSize(messageInfo.dataInfo) == 0 then
+    if helperFuncs.getTableSize(messageInfo.dataInfo) == 0 and messageInfo.ioddActive == true then
       goto nextMessage
     end
-    local localEventName = "OnNewReadMessage_" .. processingParams.port .. '_' .. messageName
+    local localEventName = "OnNewReadMessage_" .. multiIOLinkSMIInstanceNumberString .. '_' .. processingParams.port .. '_' .. messageName
+    local localEventName2 = "OnNewRawReadMessage_" .. multiIOLinkSMIInstanceNumberString .. '_' .. processingParams.port .. '_' .. messageName
     local crownEventName = "CSK_MultiIOLinkSMI." .. localEventName
+    local crownEventName2 = "CSK_MultiIOLinkSMI." .. localEventName2
 
     local function readTheMessage()
       if not processingParams.active then
-        Script.notifyEvent(localEventName, false, ioddReadMessagesQueue:getSize(), 0,  nil, 'IOLink port is not active')
+        Script.notifyEvent(localEventName, false, ioddReadMessagesQueue:getSize(), 0,  nil, 'IO-Link port is not active')
         return
       end
       local timestamp1 = DateTime.getTimestamp()
-      local success, jsonMessageContent = readIODDMessage(messageName)
-      ioddReadMessagesResults[messageName] = success
-      ioddLatestReadMessages[messageName] = jsonMessageContent
+      local success
+      local messageContent = ''
+      if messageInfo.ioddActive == false then
+        success, messageContent = readMessageNoIODD(messageName)
+      else
+        success, messageContent = readIODDMessage(messageName)
+        ioddReadMessagesResults[messageName] = success
+        ioddLatestReadMessages[messageName] = jsonMessageContent
```

Comment:
```suggestion
        ioddLatestReadMessages[messageName] = messageContent
```
Otherwise the refreshing live payload button doesn't work

Reply:
merged